core/src/main/resources/error/error-classes.json

```diff
@@ -63,5 +63,86 @@
+  "INSERT_OVERWRITE_DIRECTORY_UNSUPPORTED" : {
+    "message" : [ "INSERT OVERWRITE DIRECTORY is not supported" ],
+    "sqlState" : "10002"
+  },
+  "COLUMNS_ALIASES_NOT_ALLOWED_IN_OPERATION" : {
+    "message" : [ "Columns aliases are not allowed in %s." ],
+    "sqlState" : "10003"
+  },
+  "EMPTY_SOURCE_FOR_MERGE" : {
+    "message" : [ "Empty source for merge: you should specify a source table/subquery in merge." ],
+    "sqlState" : "10004"
+  },
+  "UNRECOGNIZED_MATCHED_ACTION" : {
+    "message" : [ "Unrecognized matched action: %s" ],
+    "sqlState" : "10005"
+  },
+  "INSERTED_VALUE_NUMBER_NOT_MATCH_FIELD_NUMBER" : {
+    "message" : [ "The number of inserted values cannot match the fields." ],
+    "sqlState" : "10006"
+  },
+  "UNRECOGNIZED_NOT_MATCHED_ACTION" : {
+    "message" : [ "Unrecognized not matched action: %s" ],
+    "sqlState" : "10007"
+  },
+  "MERGE_STATEMENT_WITHOUT_WHEN_CLAUSE" : {
+    "message" : [ "There must be at least one WHEN clause in a MERGE statement" ],
+    "sqlState" : "10008"
+  },
+  "NON_LAST_MATCHED_CLAUSE_OMIT_CONDITION" : {
+    "message" : [ "When there are more than one MATCHED clauses in a MERGE statement, only the last MATCHED clause can omit the condition." ],
+    "sqlState" : "10009"
+  },
+  "NON_LAST_NOT_MATCHED_CLAUSE_OMIT_CONDITION" : {
+    "message" : [ "When there are more than one NOT MATCHED clauses in a MERGE statement, only the last NOT MATCHED clause can omit the condition." ],
+    "sqlState" : "10010"
+  },
+  "EMPTY_PARTITION_KEY" : {
+    "message" : [ "Found an empty partition key '%s'." ],
+    "sqlState" : "10011"
+  },
+  "COMBINATION_QUERY_RESULT_CLAUSES_UNSUPPORTED" : {
+    "message" : [ "Combination of ORDER BY/SORT BY/DISTRIBUTE BY/CLUSTER BY is not supported" ],
+    "sqlState" : "10012"
+  },
+  "DISTRIBUTE_BY_UNSUPPORTED" : {
+    "message" : [ "DISTRIBUTE BY is not supported" ],
+    "sqlState" : "10013"
+  },
+  "TRANSFORM_NOT_SUPPORT_QUANTIFIER" : {
+    "message" : [ "TRANSFORM does not support DISTINCT/ALL in inputs" ],
+    "sqlState" : "10014"
+  },
+  "TRANSFORM_WITH_SERDE_UNSUPPORTED" : {
+    "message" : [ "TRANSFORM with serde is only supported in hive mode" ],
+    "sqlState" : "10015"
+  },
+  "LATERAL_WITH_PIVOT_IN_FROM_CLAUSE_NOT_ALLOWED" : {
+    "message" : [ "LATERAL cannot be used together with PIVOT in FROM clause" ],
+    "sqlState" : "10016"
+  },
+  "LATERAL_JOIN_WITH_NATURAL_JOIN_UNSUPPORTED" : {
+    "message" : [ "LATERAL join with NATURAL join is not supported" ],
+    "sqlState" : "10017"
+  },
+  "LATERAL_JOIN_WITH_USING_JOIN_UNSUPPORTED" : {
+    "message" : [ "LATERAL join with USING join is not supported" ],
+    "sqlState" : "10018"
+  },
+  "UNSUPPORTED_LATERAL_JOIN_TYPE" : {
+    "message" : [ "Unsupported LATERAL join type %s" ],
+    "sqlState" : "10019"
+  },
+  "INVALID_LATERAL_JOIN_RELATION" : {
+    "message" : [ "LATERAL can only be used with subquery" ],
+    "sqlState" : "10020"
   }
-}
\ No newline at end of file
+}
```

Comment:
This newline may be causing the test failures; we guarantee that the spacing is correct with a round trip read/write in the unit tests.